Filter by a range of `TrustThreshold` values (#2839)
informalsystems/hermes: .changelog/unreleased/improvements/ibc-relayer/2798-filter-by-trust-threshold.md

> - Filter policies now filter out client connections based on a range of trust threshold values;
>   trust thresholds within the range of [1/3, 2/3] inclusive are allowed

Reviewer: ```suggestion
- Filter policy now allows clients with trust thresholds in the range [1/3, 2/3] (inclusive)
```
